### DISCUSSION TITLE
Custom Diameq Distribution Function

### DISCUSSION THREAD

**Discussion by whishei** [neperfepx/neper#165] [Q&A] [answered]
Hi,

I am having an issue trying to generate a large grain tessellation when I give an input file for the diameq distribution. The tessellation works for 100 and 125 grain, but fails for anything 150 grains and up. Do you have a sample input file format for the distribution function? Here is an image of the probability density function and I have attached the Prob.txt file for the distribution function. 

```
$ neper -T -n 150 -morpho "diameq:file(Prob.txt)" -statcell diameq -o Test150

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.1.2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt
Info   : Running on 6 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2020, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/swhitman/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 150 -morpho diameq:file(Prob.txt) -statcell diameq
         -o Test150
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :     [i] Parsing file `Prob.txt'...
Info   :     [i] Parsed file `Prob.txt'.
Aborted (core dumped)
```

![ProbabilityDensityFunc](https://user-images.githubusercontent.com/51369978/113316853-05030600-92cc-11eb-9a7a-cfd7fb0565c2.png)

[Prob.txt](https://github.com/rquey/neper/files/6244636/Prob.txt)

**Comment by rquey** [comment, marked as answer]
The current command is
```
$ neper -T -n 150 -morpho "diameq:custom(Prob.txt)" -statcell diameq -o Test150
```

With your command, Neper was trying to read individual diameq values in `Prob.txt`, which is why it failed for `-n` larger than the number of entries in the file.

2 advices:
- specify a sphericity distribution (e.g. the default) to avoid getting weird grain shapes.
- smooth your distribution (although Neper takes it just fine).

**Reply by whishei** [reply]
Thank you!
